### DISCUSSION TITLE
General discussion and feedback

### DISCUSSION THREAD

**Discussion by yermak** [yermak/AudioBookConverter#44] [General]
**Update:** discussions are moved here https://github.com/yermak/AudioBookConverter/discussions

> Dear users,
> 
> if you have an idea or any feedback on program usability features, please share it here and i will pick it up as separate issue for future development.

**Comment by Kazaa3** [comment]
Hallo yermak,
i will extend this post soon. As first suggesion for features:

- Presets for chapter naming
- A settings menu for default values (like bitrate)
- ALAC/FLAC support
- Opening of already created m4b to edit menu and add files
- If you want to keep the GUI simple, a basic and an "expert" mode

**Comment by Kazaa3** [comment]
At the moment many steps are necessary to create a m4b-file with menu:

1. Drop Files
2. Click on a file & Select all (Ctrl+A)
3. Click on Import
4. Click on Chapter 1
5. Click on Edit
6. Make menu setings
7. Check Apply for all chapters
8. Click on OK
9. Click on Convert
10. Click on Start

Wouldn't a "Chapter tab" be more comfortable:
![grafik](https://user-images.githubusercontent.com/43350149/77906699-a9510780-7288-11ea-8d07-b95b0558a975.png)
Which leads directly to this screen:
![9](https://user-images.githubusercontent.com/43350149/77907706-5ed08a80-728a-11ea-9ccd-b00e9c3cb089.PNG)


At the moment the settings have to be set for each file individually, how about saving settings. The same goes for chapter naming.
![Presets](https://user-images.githubusercontent.com/43350149/77921133-a7467300-729f-11ea-926d-ae041ed5e601.PNG)

**Comment by yermak** [comment]
> At the moment many steps are necessary to create a m4b-file with menu:
> 
> 1. Drop Files
> 2. Click on a file & Select all (Ctrl+A)
> 3. Click on Import
> 4. Click on Chapter 1
> 5. Click on Edit
> 6. Make menu setings
> 7. Check Apply for all chapters
> 8. Click on OK
> 9. Click on Convert
> 10. Click on Start
> 
> Wouldn't a "Chapter tab" be more comfortable.

It's currently implemented exactly with "Chapter tab", I just hide one and show another. 
I don't see easy way to keep them in sync... if moved or removed files/ how to reflect in chapters? And vice versa?

However, point is taken... that's why I'm against any expert modes...
Few steps could be optimized:
>3. Click on Import

In theory could do autoimport to chapters, however, if you order of files is wrong - it's more difficult to order them in chapters modes (could be setting flag though)
>5.  Click on Edit
>6. Make menu setings
>7. Check Apply for all chapters
> 8. Click on OK

Even before presets could save user default setting for chapters naming.

>?? Output settings

Also could save default user settings.

>9 Click on Convert

Yep, this could be removed completely - and keep just set of buttons.

Presets... is it too many settings? what about program level profiles? I.e. one profile which contain all settings, you can mix and match profile to have equalent of different presets...

**Comment by yermak** [comment]
About autoimport... i will require some optimisation, if you noticed import does not start before all files are scanned - quite long time on large books, if i do autoimport it could be done in background...

**Comment by yermak** [comment]
> * ALAC/FLAC support

Do you mean for import or as output? It seems an overkill for audiobooks....

**Comment by yermak** [comment]
> * If you want to keep the GUI simple, a basic and an "expert" mode

Not sure about "expert" mode.... goal is any user can add several mp3 files w/o extra knowledge get resulting audiobook.

**Comment by yermak** [comment]
> * Opening of already created m4b to edit menu and add files

Why this would be needed by someone?

**Comment by TheMetalCenter** [comment]
Thanks for the latest update! I’d also think both this and “presets” for chapters would be great editions. I’m imagining presets as copying already existing metadata into the chapter name. For example, set chapter names to <filename> of track. Or set to <title> of track. This would save a lot of time on large books with 100+ chapters. 

As for opening already created m4b’s, I think this is more limited use case but occasionally I have an audiobook with a few m4b files (say 3) that have already existing chapter metadata (say 5 each). Currently, if you import these to the program you will get a new m4b with only three chapters, rather than 15. Existing chapters metadata is lost.

**Comment by Kazaa3** [comment]
> 
> 
> > * ALAC/FLAC support
> 
> Do you mean for import or as output? It seems an overkill for audiobooks....

Yes, as input files to convert them directly into m4a/m4b acc (like for mp3 files).

**Comment by yermak** [comment]
@siteuser42 
> For example, set chapter names to <filename> of track. Or set to <title> of track. This would save a lot of time on large books with 100+ chapters.

Just to make sure - this is currently available and you could do this with few clicks, but preset will help with configuring.

> if you import these to the program you will get a new m4b with only three chapters, rather than 15. Existing chapters metadata is lost.

Well, that's very interesting - re-importing meta from m4b is usefull and easy, manual alignement of chapters to timing, i.e. proper editing of chapters... will make UI complex...

**Comment by yermak** [comment]
> > > * ALAC/FLAC support
> > 
> > 
> > Do you mean for import or as output? It seems an overkill for audiobooks....
> 
> Yes, as input files to convert them directly into m4a/m4b (like for mp3 files).

As quick hack - this works just out of the box, just need add supported extensions.
Need a bit of work to process .cue file and read meta information from it.

Create issue #47. 

Or is it better just do direct rip of CD to audiobook? Not sure if it's possible yet.

What about .ogg?

**Comment by Kazaa3** [comment]
> 
> 
> @siteuser42
> 
> > For example, set chapter names to  of track. Or set to <title> of track. This would save a lot of time on large books with 100+ chapters.
> 
> Just to make sure - this is currently available and you could do this with few clicks, but preset will help with configuring.
> 
> > if you import these to the program you will get a new m4b with only three chapters, rather than 15. Existing chapters metadata is lost.
> 
> Well, that's very interesting - re-importing meta from m4b is usefull and easy, manual alignement of chapters to timing, i.e. proper editing of chapters... will make UI complex...

This is exactly the usecase that would be cool. Inserting several m4b's that are already merged and make a new m4b containing all chapters of the original files.

The second use case, is to rename the menu afterwards.The background is that I deleted the original files after the m4b was created. Now I would like to add the title names afterwards.

The third use case is: In some of my audio books the numbering is not correct, so I would like to change the order of the titles.

**Comment by yermak** [comment]
> > @siteuser42
> > > For example, set chapter names to  of track. Or set to <title> of track. This would save a lot of time on large books with 100+ chapters.
> > 
> > 
> > Just to make sure - this is currently available and you could do this with few clicks, but preset will help with configuring.
> > > if you import these to the program you will get a new m4b with only three chapters, rather than 15. Existing chapters metadata is lost.
> > 
> > 
> > Well, that's very interesting - re-importing meta from m4b is usefull and easy, manual alignement of chapters to timing, i.e. proper editing of chapters... will make UI complex...
> 
> This is exactly the usecase that would be cool. Inserting several m4b's that are already merged and make a new m4b containing all chapters of the original files.
> 
> The second use case, is to rename the menu afterwards.The background is that I deleted the original files after the m4b was created. Now I would like to add the title names afterwards.
> 
> The third use case is: In some of my audio books the numbering is not correct, so I would like to change the order of the titles.

Created issue #49

**Comment by Kazaa3** [comment]
> Or is it better just do direct rip of CD to audiobook? Not sure if it's possible yet.

What do you mean by that? By using AudioBookConverter or other software like iTunes? In my case I use EAC to rip physical media as loseless files and store them on a NAS. After ripping I'm only using the digital files.

> What about .ogg?

Each additional sound format/Container makes the application more attractive. Opus and Vorbis comment could also be interesting. Which audio formats (and extensions) are currently supported? Maybe this could be included in the documentation. Something like that:

Output Format:
m4b (M4A  (isom/iso2))

Supported input audio formats: 
m4a (acc)
m4b
mp3 
wma

Unsupported:
m4a (alac) [menus unsupported]
flac 

Which software is limiting the input file formats? FFmpeg? MP4v2? AndI have also a question about the output file format. The file format seems to be a MP4-container with M4A (isom/iso2) and .m4b extension. Their are 2 different menus written by AudioBookConverter. What are the formats of the meta data? 

My iPod classic shows only the first menu. And I don't know any application to edit them. Other apps seems to read menu 2. For example VLC. These Tags are also editable by MP3tag. Somewhere I've read the second menu meta data is called Nero tags (MP4-Nero). Is there any more information about this topic?
EDIT: The first one seems to be QuickTime chapter information.

**Comment by yermak** [comment]
In theory we can process everything supported by ffmpeg, however, each format requires testing, which makes it more difficult to maintain.

**Comment by HenryMontrose** [comment]
Would be great to have 2 modes.. i.e. simple..and advanced, because as the programme gets taller after pressing folders (organise). when you press start, instead of it starting, you then have to work through the start sequence.. for peeps like me, who want to utilise it simply i.e. add mp3 add artwork, maybe author name and 64kpbs.. to add to my ipod, a simple all in 1 would be wonderful, and of course, those flac people..even though apple doesn't support it, can play to their hearts content in advanced mode.. wonderful programme, so please you re-invented the wheel .

**Comment by TheMetalCenter** [comment]
> @siteuser42
> 
> > For example, set chapter names to  of track. Or set to <title> of track. This would save a lot of time on large books with 100+ chapters.
> 
> Just to make sure - this is currently available and you could do this with few clicks, but preset will help with configuring.
> 
I see! I hadn't gotten a chance to try out the update. Just did and indeed it worked great. Thanks again.

**Comment by HenryMontrose** [comment]
No offence meant, and your English Is wonderful, but 

>  **discussion**

 , is the correct spelling, not discission

**Comment by yermak** [comment]
Question to all: how long typically your average book is being encoded? Are you happy with program performance?

**Comment by TheMetalCenter** [comment]
Back on version 4.3 I compiled a few 50 hour books (around 100 chapters) and they worked well! I saved all the split files, so I am going to redo them with the new updates to retain chapter titles.

**Comment by HenryMontrose** [comment]
Hi , usually v quick, say 3-5 mins.. thanks 
-- 
 Securely sent with Tutanota. Get your own encrypted, ad-free mailbox: 
 https://tutanota.com


31 Mar 2020, 23:50 by notifications@github.com:

**Comment by TheMetalCenter** [comment]
The 50 hour books from mp3 files takes about half an hour at 64 bitrate

**Comment by yermak** [comment]
Hi all,
Yet another though, would it be useful to reencode m4b back to mp3 splitting by chapters?

**Comment by TheMetalCenter** [comment]
I personally don’t have an interest in this, but I know some people prefer their audiobooks that way.

**Comment by HenryMontrose** [comment]
Notice a huge lag between pressing start button and the actual title to be dropped into folder on 4.5.2 was never there before, and like your website, but you missed out a "T" on Simple and ### lightweigh user interface: convert a book with few clicks..

**Comment by yermak** [comment]
> Notice a huge lag between pressing start button and the actual title to be dropped into folder on 4.5.2 was never there before, and like your website, but you missed out a "T" on Simple and ### lightweigh user interface: convert a book with few clicks..

@HenryMontrose, that's known issue - i believe all versions from 4.0 are impacted. I assume - you did not do anything with chapters - just added files and pressed start. In version 4.x even if you don't import chapters - they are implicitely imported and conversion can only start when all chapters are processed, while in version 3.x - process could be started immidately. I don't like this behaviour as well, I will resolve at later versions. As workaround - spend more time selecting artwork  while you do this chapters will be loaded :)

**Comment by HenryMontrose** [comment]
Thanks for your help as always. 
-- 
 Securely sent with Tutanota. Get your own encrypted, ad-free mailbox: 
 https://tutanota.com


4 Apr 2020, 22:58 by notifications@github.com:

**Comment by HenryMontrose** [comment]
Love what you have done moving the conversion tab to the front ,I guess if you wanted, you could move website, FAQ, that are now partially hidden to that huge gap below AudioBookConverter 4.60

**Comment by yermak** [comment]
@HenryMontrose, could you please make a screenshot of what is hidden?

I know program may not behave well in different resolutions and different scaling in windows. 
So, if you see something particularly ugly - please let me know, I'm trying to focus on features and simplicity of use, so interface may suffer sometimes :)

**Comment by HenryMontrose** [comment]
Hi Yermak

enclosed screenshot where you will clearly see the hidden, I use a pc and is clearly hidden  i.e. website and 0 next to book info tab 
-- 
 Securely sent with Tutanota. Get your own encrypted, ad-free mailbox: 
 https://tutanota.com


11 Apr 2020, 17:00 by notifications@github.com:

**Comment by HenryMontrose** [comment]
> Hi Yermak enclosed screenshot where you will clearly see the hidden, I use a pc and is clearly hidden  i.e. website and 0 next to book info tab  -- Securely sent with Tutanota. 
![Screenshot (2)](https://user-images.githubusercontent.com/61348379/79074856-a7566200-7ce6-11ea-825e-731dc2e6b218.png)

**Comment by yermak** [comment]
@HenryMontrose , I understand where issue is, you either using low resolution screen or high percentage of scaling. Unfortunately, i did not put enough efforts to make it work on all possible resolutions and scales, will work on it, as you may be not along.

As a temporary workaround, you could go to "Display Settings" and change "Scale" to lower percentage to make it work.

I really suspended all other ideas to fix UI asap.

**Comment by HenryMontrose** [comment]
Just thought I would let dust settle on 5.1, I tried it today, a good improvement, just off a little on height, but not too bad, the UI is a lot better and clearer , still a bit of lag between pressing start and it starting.. but as always thanks for the work .

**Comment by yermak** [comment]
Version 5.2.0 is out. any feedback?

**Comment by HenryMontrose** [comment]
Sorry, will download it today and report back to you, best 
-- 
 Securely sent with Tutanota. Get your own encrypted, ad-free mailbox: 
 https://tutanota.com


26 May 2020, 09:42 by notifications@github.com:

**Comment by HenryMontrose** [comment]
Gave it a serious workout, added15 books in a row, didn't realise you could batch convert, worked flawlessly , all I would like to know is that even when I merge all the chapters, and it converts into 1 , when it drops into my ipod touch, it seems to go  back to  into chapters (A), what I like is (B) all in one..
Great work 
Best regards
-- 
 Securely sent with Tutanota. Get your own encrypted, ad-free mailbox: 
 https://tutanota.com


26 May 2020, 09:42 by notifications@github.com:

**Comment by lou2k7** [comment]
Awesome program. works great. Love being able to put in the series name and other fields.

When I put in book#  '01'  the output file name removes the 0 and outputs as 1. Is there a way to make it include the begin zero.

also. I'm trying to output the year in the name.  by adding <if(YEAR)><YEAR> <endif> to the end. Doesnt work for me. figure i was doing something wrong. 
filename_format=<WRITER> <if(SERIES)>- [<SERIES><if(BOOK_NUMBER)> -<BOOK_NUMBER><endif>] <endif>- <TITLE><if(NARRATOR)> (<NARRATOR>)<endif><if(YEAR)><YEAR><endif>

thank you again

**Comment by yermak** [comment]
@lou2k7,
thank you for feedback.
As for numbering with leading zero - this relates to issue #87 - meaning I need to stop storing book numbers as integer numbers, this is possible, but those numbers wont be updated in media info.
As for using YEAR in templates - this could be considered as bug - I never bothered to pass YEAR into template. This will be fixed in next version, which is considering new lockdown announced in the UK and X-mas comming should be soon.

**Comment by red-ochre** [comment]
I have multiple .m4a files that are already in the quality that I need and I want to combine them into a .m4b with chapters but without reencoding. Can that be done? I can't see in the "quality" tab how I can tell it to just pass through.

**Comment by yermak** [comment]
> I have multiple .m4a files that are already in the quality that I need and I want to combine them into a .m4b with chapters but without reencoding. Can that be done? I can't see in the "quality" tab how I can tell it to just pass through.

It depends what codec is used inside yours m4a. If codec is aac then there will be no reencoding (in case you selected m4b or m4a as output) - you could notice  it by very fast processing, however m4a can contain apple alac codec, then program will reencode files according to selected settings.

**Comment by yermak** [comment]
Hi all,
Github finally introduced new feature - Discussions.
Please share your feedback, ideas and ask questions here.

**https://github.com/yermak/AudioBookConverter/discussions**

**Comment by red-ochre** [comment]
> 
> It depends what codec is used inside yours m4a. If codec is aac then there will be no reencoding (in case you selected m4b or m4a as output) - you could notice it by very fast processing, however m4a can contain apple alac codec, then program will reencode files according to selected settings.

Thx Yermak.
Just a suggestion to make it easier for users to see what will happen, maybe, we could see on the "quality" tab if it will simply be a pass-through process? It's just that I might want a particular set of parameters and I set them on the "quality" tab but then the process changes it to pass-through without me knowing about it. Or, the opposite, where I might expect it to just pass-through but the system decides that it can't do that and goes with the settings on the quality tab.

**Comment by d-rebs** [comment]
I have a feature request to create sub directories or save to said directories based on WRITER, SERIES, etc. I've been using Calibre to organize but I hate how it makes a copy of the audiobooks. Ideally I'd love it if this could save like 
`filename_format=<BOOK_NUMBER> - <TITLE>`
`output.folder=D\:\\Audiobooks\\<WRITER> - <SERIES>\\`

**Comment by randecse** [comment]
I'm using v5.5.26 and the temporary M4B files don't get deleted after completing a project. They files stay in AppData\Local\Temp after rebooting the PC too.
